### ISSUE TITLE
Startup: [Mac IDE] GM crashes in 2024.1100.0.634 due to new IPv6/4 routing fix

### ISSUE THREAD

**Issue by yerumaku** [YoYoGames/GameMaker-Bugs#7894]
### Description
It's just that at some point, when I didn't even press anything, an instability message appeared and the application was closed.

### Which version of GameMaker are you reporting this issue for?
IDE v2024.1100.0.634 Runtime v2024.1100.0.658

### Which operating system(s) are you seeing the problem on?
Mac 14.6.1

[e34c3ccc-5c6a-44ca-b893-9358ff14735b](https://api.gamemaker.io/api/github/downloads/e34c3ccc-5c6a-44ca-b893-9358ff14735b)

**Comment by YYDan** [comment]
Looks to be that your system does not like the fix for https://github.com/YoYoGames/GameMaker-Bugs/issues/7152

The ui.log contains loads of giant callstacks when attempting to reach various/all webhosts:

```
[14:16:51:056(b613)] --------------------------------------------------------------------------------
[14:16:51:056(b613)] Unobserved Task Exception: System.Net.Sockets.SocketException
[14:16:51:056(b613)] Exception message: No route to host
[14:16:51:056(b613)] Exception stack:    at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.CreateException(SocketError error, Boolean forAsyncThrow)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ConnectAsync(Socket socket)
   at System.Net.Sockets.Socket.ConnectAsync(EndPoint remoteEP, CancellationToken cancellationToken)
   at YoYoStudio.Core.Utils.HttpUtil.HappyEyeballsHttp.AttemptConnection(Int32 index, IPAddress address, Int32 port, CancellationToken cancel)
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at YoYoStudio.Core.Utils.HttpUtil.HappyEyeballsHttp.AttemptConnection(Int32 index, IPAddress address, Int32 port, CancellationToken cancel)
   at YoYoStudio.Core.Utils.HttpUtil.HappyEyeballsHttp.<>c__DisplayClass2_0.<OnConnect>b__0(Int32 i, CancellationToken cancel)
   at YoYoStudio.Core.Utils.HttpUtil.HappyEyeballsHttp.ParallelTask[T](Int32 candidateCount, Func`3 taskBuilder, TimeSpan delay, CancellationToken cancel)
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at YoYoStudio.Core.Utils.HttpUtil.HappyEyeballsHttp.ParallelTask[T](Int32 candidateCount, Func`3 taskBuilder, TimeSpan delay, CancellationToken cancel)
   at YoYoStudio.Core.Utils.HttpUtil.HappyEyeballsHttp.OnConnect(SocketsHttpConnectionContext context, CancellationToken cancellationToken)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.ExecutionContextCallback(Object s)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext(Thread threadPoolThread)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext()
   at System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox box, Boolean allowInlining)
   at System.Threading.Tasks.Task.RunContinuations(Object continuationObject)
   at System.Threading.Tasks.Task`1.TrySetResult(TResult result)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetExistingTaskResult(Task`1 task, TResult result)
   at YoYoStudio.Core.Utils.HttpUtil.HappyEyeballsHttp.GetIpsForHost(DnsEndPoint endPoint, CancellationToken cancel)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.ExecutionContextCallback(Object s)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext(Thread threadPoolThread)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext()
   at System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox box, Boolean allowInlining)
   at System.Threading.Tasks.Task.RunContinuations(Object continuationObject)
   at System.Threading.Tasks.Task.FinishSlow(Boolean userDelegateExecute)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart()
   at System.Threading.Thread.StartCallback()
--- End of stack trace from previous location ---
   at YoYoStudio.Core.Utils.HttpUtil.HappyEyeballsHttp.AttemptConnection(Int32 index, IPAddress address, Int32 port, CancellationToken cancel)
[14:16:51:056(b613)] --------------------------------------------------------------------------------
```

Will need to investigate the original change further.

**Comment by yoyo-danielc** [comment]
The "No route to host" exception gets raised on Linux and macOS when there is no IPv6 connectivity on the system, I've put a fix on the 2024.11 branch, please re-open this issue if it isn't resolved for you in the next beta.

**Comment by Emc1923** [comment]
Closing as per the above comment.
